a bit of this

vitalii-monocl opened the pull request:
a bit of that

vitalii-monocl committed:
add requirements.txt

vitalii-monocl committed:
remove wrong module from requirements.txt

vitalii-monocl committed:
keep empty "static" dir in git repo, ignore it's actual contents in the future

vitalii-monocl committed:
ignore python's cache and virtual env

vitalii-monocl committed:
comment for requirements installation

vitalii-monocl committed:
exclude pdf_reader_settings from git repo

vitalii-monocl committed:
put results in the "output" directory, create one if doesn't exist

vitalii-monocl committed:
keep output out of the repo

vitalii-monocl committed:
merge

PalibrixBeet commented:
@vitalii-monocl , just curious why do we need empty static folder on git. Newer saw that)

And maybe we can reduce requirements. I think we need only PyMuPDF, pdfplumber and alive-progress

vitalii-monocl commented:
Honestly, I have no idea what would be optimal in terms of requirements. I just dropped everything I saw in chat earlier. And made it work.

As for that empty dir "static", it was required by that "fitz" module. Refused to run without it. I thought you forgot about git ignoring empty dirs and so on.
And later I figured out it was wrong fitz in that requirements.txt, my bad.

P.S. I'll do another pull request with "static" removed and also I added "output" dir, with .gitignore and edited some of your code to write files in that output dir, instead of main one. That's useful, or am I confused again?

vitalii-monocl committed:
remove empty "static" dir (it was required by wrong fitz module, we don't need it)

PalibrixBeet commented:
Right now, it's storing output in the same directory where pdf is located. In the future - I would make that user might select where exactly the file must be stored. But for now I decided not to make it too complex - I don't even know if it will be useful

I think it's not that bad of a change

PalibrixBeet committed:
Update .gitignore

PalibrixBeet commented:
Can we close it now, or any more changes incoming?

vitalii-monocl commented:
My idea was to make it a bit more handy to use it and develop it in the same place. Just my preference, not sure if that is even a good practice or not.

No more changes from me, time to go to sleep :)

PalibrixBeet commented:
Yeah, that's definitely better. I completely forgot about existence of those files). Thanks. I'll add README.md later
